vsphere_plugin_common/__init__.py

```diff
@@ -1058,6 +1118,10 @@ def create_server(self,
             vm_memory=memory,
         )
 
+        # Correct the network name for all networks from relationships
+        for network in networks:
+            network['name'] = self._get_connected_network_name(network)
```

Comment:
Validate inputs provides helpful output to the user beforehand so that everything else can assume it works, and the _get_connected_etc_etc call doesn't make any extra calls to the platform so doing it twice I think is reasonable given the intent to provide better error messages?